.github/workflows/regression_tests.yml

```diff
@@ -45,25 +44,25 @@ jobs:
         id: check_changes
         run: echo "has_changes=${{ steps.changes.outputs.has_changes }}" >> $GITHUB_ENV
 
-      - name: Set up Python ${{ matrix.python-version }}
+      - name: Set up Python 3.10
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: '3.10'
 
       - name: Restore Python environment cache
         if: env.has_changes == 'true'
         id: restore-env
         uses: actions/cache/restore@v4
         with:
-          path: .venv-${{ matrix.python-version }}
-          key: ${{ runner.os }}-venv-${{ matrix.python-version }}-${{ hashFiles('devtools/dev-requirements.txt', 'requirements.txt') }}
```

Comment:
instead of using `matrix.python-version` as the key why can't we use the actual version from `python --version`

Reply:
Btw this problem will probably be solved when every Github runner starts using 3.12.8 and 3.11.11. For now, some of them use 3.12.7 and this causes the issue. They made some security changes regarding the `venv` activation with the latest patch